services/grpc/grpc.js

```diff
@@ -141,12 +142,25 @@ class Grpc extends EventEmitter {
    * Rejig connections as needed before activating the lightning service.
    */
   async onBeforeActivateLightning() {
-    await this.services.Lightning.connect()
+    const { Lightning } = this.services
+    await Lightning.connect()
+    // creates listener that re-emits specified event
+    const forwardEvent = event => data => this.emit(event, data)
+    Lightning.on('subscribeInvoices.data', forwardEvent('subscribeInvoices.data'))
+    Lightning.on('subscribeTransactions.data', forwardEvent('subscribeTransactions.data'))
+    Lightning.on('subscribeChannelGraph.data', forwardEvent('subscribeChannelGraph.data'))
   }
+
   async onAfterActivateLightning() {
     this.emit('GRPC_LIGHTNING_SERVICE_ACTIVE')
   }
 
+  onLeaveActive() {
+    this.removeAllListeners('subscribeInvoices.data')
```

Comment:
Shouldn't this be removing listeners for ` this.services.Lightning` that we added here, rather than removing listeners that users of the `Grpc` class have added?

Reply:
I was thinking about that too. But didn't want to introduce instance variables to track event handlers. My reasoning was that particular Lightning variable is an internal instance and we fully control those listeners and should not be having more than one handler per event. If you think this is confusing i can refactor. Also in a future update (the discussed refactor of grpc) Lightning won't get exposed at all, so there will be no chance client code adds its own listeners.

Reply:
There is no need to track individual handlers and remove one by one. You can just use `removeAllListeners` as you are already:

```
this.removeAllListeners('subscribeInvoices.data')
this.removeAllListeners('subscribeTransactions.data')
this.removeAllListeners('subscribeChannelGraph.data')
```
->
```
this.services.Lightning.removeAllListeners('subscribeInvoices.data')
this.services.Lightning.removeAllListeners('subscribeTransactions.data')
this.services.Lightning.removeAllListeners('subscribeChannelGraph.data')
```

I'm just suggesting that you remove the event forwarding listeners that you added to `this.services.Lightning` in `onBeforeActivateLightning`, rather than removing listeners to `this` that external users of the class added.

Reply:
Yeah, you are 100% right. My bad, I've misread your original comment